translateCoverage.py

```diff
@@ -3,103 +3,8 @@
-        dst_rule_map[dst_rule_loc.strip()] = dst_rule_hash.strip()
-
-    src_rule_list = [ rule_hash.strip() for rule_hash in src_rules_list ]
-
-    # Filter out non-functional rules from rule map (determining if they are functional via the top symbol in the rule being `<generatedTop>`)
-    dst_non_function_rules = []
-    dst_kompiled = pyk.readKastTerm(dst_kompiled_dir + '/compiled.json')
-    for module in dst_kompiled['modules']:
-        for sentence in module['localSentences']:
-            if pyk.isKRule(sentence):
-                ruleBody = sentence['body']
-                ruleAtt  = sentence['att']['att']
-                if    (pyk.isKApply(ruleBody)                                     and ruleBody['label']        == '<generatedTop>') \
-                   or (pyk.isKRewrite(ruleBody) and pyk.isKApply(ruleBody['lhs']) and ruleBody['lhs']['label'] == '<generatedTop>'):
-                    if 'UNIQUE_ID' in ruleAtt:
-                        dst_non_function_rules.append(ruleAtt['UNIQUE_ID'])
-
-    # Convert the src_coverage rules to dst_no_coverage rules via the maps generated above
-    dst_rule_list = []
-    for src_rule in src_rule_list:
-        if src_rule in src_rule_map:
-            src_rule_loc = src_rule_map[src_rule]
-            if src_rule_loc in dst_rule_map:
-                dst_rule = dst_rule_map[src_rule_loc]
-                if dst_rule in dst_non_function_rules:
-                    dst_rule_list.append(dst_rule)
-                else:
-                    _notif('Skipping non-semantic rule: ' + dst_rule)
-            else:
-                _fatal('COULD NOT FIND RULE LOCATION IN dst_rule_map: ' + src_rule_loc)
-        else:
-            _fatal('COULD NOT FIND RULE IN src_rule_map: ' + src_rule)
-
-    return dst_rule_list
-
-def translateCoverageFromPaths(src_komplied_dir, dst_kompiled_dir, src_rules_file):
-    """Translate coverage information given paths to needed files.
-
-    Input:
-
-        -   src_kompiled_dir: Path to *-kompiled directory of source.
-        -   dst_kompiled_dir: Path to *-kompiled directory of destination.
-        -   src_rules_file: Path to generated rules coverage file.
-
-    Output: Translated list of rules with non-semantic rules stripped out.
-    """
-    src_all_rules = []
-    with open(src_kompiled_dir + '/allRules.txt', 'r') as src_all_rules_file:
-        src_all_rules = [ line.strip() for line in src_all_rules_file ]
-
-    dst_all_rules = []
-    with open(dst_kompiled_dir + '/allRules.txt', 'r') as dst_all_rules_file:
-        dst_all_rules = [ line.strip() for line in dst_all_rules_file ]
-
-    dst_definition = pyk.readKastTerm(dst_kompiled_dir + '/compiled.json')
-
-    src_rules_list = []
-    with open(src_rules_file, 'r') as src_rules:
-        src_rules_list = [ line.strip() for line in src_rules ]
-
-    return translateCoverage(src_all_rules, dst_all_rules, dst_definition, src_rules_list)
+from pykWasm      import *
+from pyk.coverage import *
```

Comment:
Generally, glob imports like these are to be avoided.

Reply:
Well I figured since these are libraries I control (one being in the same repo here, one being maintained soley by me and being fairly new), that it would be safe here.

Reply:
Well I figured since these are libraries I control (one being in the same repo here, one being maintained soley by me and being fairly new), that it would be safe here.